ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -390,57 +390,44 @@ public Map<String, CloudBlobMetadata> getBlobMetadata(List<BlobId> blobIds) thro
   @Override
   public CompletableFuture<Map<String, CloudBlobMetadata>> getBlobMetadataAsync(List<BlobId> blobIds) {
     Objects.requireNonNull(blobIds, "blobIds cannot be null");
-    CompletableFuture<Map<String, CloudBlobMetadata>> resultFuture = new CompletableFuture<>();
-
     if (blobIds.isEmpty()) {
-      resultFuture.complete(Collections.emptyMap());
-      return resultFuture;
+      return CompletableFuture.completedFuture(Collections.emptyMap());
     }
 
     // For single blob GET request (isVcr = false), get metadata from ABS instead of Cosmos
     // Note: findMissingKeys (isVcr = true) needs to query Cosmos regardless, because a partial failure in a
     // previous upload could have resulted in a missing record in Cosmos; the findMissingKeys result
     // needs to include that store key to replay the upload.
     if (!isVcr && blobIds.size() == 1) {
-      azureBlobDataAccessor.getBlobMetadataAsync(blobIds.get(0)).whenComplete((cloudBlobMetadata, throwable) -> {
+      return azureBlobDataAccessor.getBlobMetadataAsync(blobIds.get(0)).handle((cloudBlobMetadata, throwable) -> {
         if (throwable != null) {
           Exception ex = Utils.extractFutureExceptionCause(throwable);
-          resultFuture.completeExceptionally(
+          throw new CompletionException(
               toCloudStorageException("Failed to query metadata for blob" + blobIds.get(0), ex));
         } else {
           if (cloudBlobMetadata == null) {
-            resultFuture.complete(Collections.emptyMap());
+            return Collections.emptyMap();
           } else {
-            resultFuture.complete(Collections.singletonMap(cloudBlobMetadata.getId(), cloudBlobMetadata));
+            return Collections.singletonMap(cloudBlobMetadata.getId(), cloudBlobMetadata);
```

Comment:
though I think this can be as simple as 
```
return azureBlobDataAccessor.getBlobMetadataAsync(blobIds.get(0)).thenApply(md -> md == null? Collections.emptyMap()? Collections.singletonMap(md.getId(), md));
```